src/index.ts

```diff
@@ -185,45 +244,107 @@ export async function decryptWithKey<R>(
+    return {
+      key: await window.crypto.subtle.importKey(
+        EXPORT_FORMAT,
+        exportedEncryptionKey.key,
+        DERIVED_KEY_FORMAT,
+        true,
+        ['encrypt', 'decrypt'],
+      ),
+      derivationOptions: exportedEncryptionKey.derivationOptions,
+    };
+  }
+
+  return await window.crypto.subtle.importKey(
     EXPORT_FORMAT,
-    JSON.parse(keyString),
+    exportedEncryptionKey,
     DERIVED_KEY_FORMAT,
     true,
     ['encrypt', 'decrypt'],
   );
-
-  return key;
 }
 
 /**
  * Receives an exported CryptoKey string, creates a key,
  * and decrypts cipher text with the reconstructed key.
  *
- * @param key - The CryptoKey to export.
+ * @param encryptionKey - The CryptoKey to export.
  * @returns A key string.
  */
-export async function exportKey(key: CryptoKey): Promise<string> {
-  const exportedKey = await window.crypto.subtle.exportKey(EXPORT_FORMAT, key);
-  return JSON.stringify(exportedKey);
+export async function exportKey(
+  encryptionKey: CryptoKey | EncryptionKey,
+): Promise<string> {
+  if (isEncryptionKey(encryptionKey)) {
+    return JSON.stringify({
+      key: await window.crypto.subtle.exportKey(
+        EXPORT_FORMAT,
+        encryptionKey.key,
+      ),
+      derivationOptions: encryptionKey.derivationOptions,
+    });
+  }
+
+  return JSON.stringify(
+    await window.crypto.subtle.exportKey(EXPORT_FORMAT, encryptionKey),
+  );
 }
 
 /**
  * Generate a CryptoKey from a password and random salt.
  *
  * @param password - The password to use to generate key.
  * @param salt - The salt string to use in key derivation.
- * @param exportable - Should the derived key be exportable.
+ * @param exportable - Whether or not the key should be exportable.
+ * @returns A CryptoKey for encryption and decryption.
+ */
+export async function keyFromPassword(
+  password: string,
+  salt: string,
+  exportable?: boolean,
+): Promise<CryptoKey>;
+/**
+ * Generate a CryptoKey from a password and random salt, specifying
+ * key derivation options.
+ *
+ * @param password - The password to use to generate key.
+ * @param salt - The salt string to use in key derivation.
+ * @param exportable - Whether or not the key should be exportable.
+ * @param opts - The options to use for key derivation.
+ * @returns A CryptoKey for encryption and decryption.
+ */
+export async function keyFromPassword(
+  password: string,
+  salt: string,
+  exportable?: boolean,
+  opts?: KeyDerivationOptions,
+): Promise<EncryptionKey>;
+/**
+ * Generate a CryptoKey from a password and random salt, optionally
+ * specifying key derivation options.
+ *
+ * When no derivation options are specified, the default options are used.
+ *
+ * @param password - The password to use to generate key.
+ * @param salt - The salt string to use in key derivation.
+ * @param exportable - Whether or not the key should be exportable.
+ * @param opts - The options to use for key derivation.
  * @returns A CryptoKey for encryption and decryption.
  */
```

Comment:
Nit: This is the implementation of the function, but doesn't need JSDoc above it. It doesn't hurt to have this here, but VSCode (or whatever is reading this) will get the JSDoc for this function from the overloads.

That said, I realize that removing these lines currently produces a lint error. I've commonly gotten around this by just adding an ESLint ignore:

``` suggestion
// The overloads are already documented.
// eslint-disable-next-line jsdoc/require-jsdoc
```